src/main/docs/guide/httpServer/serverConfiguration/listener.adoc

```diff
@@ -43,3 +43,48 @@ micronaut:
 - `unixListener` is a listener name, and can be an arbitrary value
 
 NOTE: To use an abstract domain socket instead of a normal one, prefix the path with a NUL character, like `"\0/run/micronaut.socket"`
+
+== systemd socket activation support
+
+When using the epoll transport, the HTTP server can be configured to use an existing file descriptor. With this feature, you can use socket created by systemd and passed to the micronaut process:
+
+[configuration]
+----
+micronaut:
+  netty:
+    event-loops:
+      # use epoll
+      parent:
+        prefer-native-transport: true
+      default:
+        prefer-native-transport: true
+    listeners:
+      systemd:
+        fd: 3 # systemd passes the server socket as fd 3
```

Comment:
Is this always `3`?

Or is this dependant on when systemd sets up the socket via the below config?

Are you on a Mac?  If so, how did you test this? ;-)

Reply:
it's always 3. and no, i'm on linux.

Reply:
https://github.com/systemd/systemd/blob/main/src/systemd/sd-daemon.h#L56

Reply:
can we not default the value to 3 if it is always 3?

Reply:
i think it could be 0 if you were using inetd for example. the 3 is systemd-specific, but the fd config option is more generic.

Reply:
ive thought about detecting it from the env variables that systemd sets. but it would be a bit more complicated of a PR, there's no way to test it, it might have negative perf impact, and it is pretty niche in the first place.

this pr is basically the minimum to make it work with systemd with config options only, with no adverse impact since it's a niche feature.

Reply:
> https://github.com/systemd/systemd/blob/main/src/systemd/sd-daemon.h#L56

If I'm reading that right, it starts at 3, but if you have multiple services, they'll be on `3 + n`?

Reply:
right. i dont really see a use case for micronaut there though, so i didnt add it to the docs.